imaginairy/schema.py

```diff
@@ -172,385 +133,543 @@ def handle_b64(value: Any) -> "LazyLoadingImage":
+    # @field_validator("mask_prompt", "mask_image", mode="before")
+    # def validate_mask_prompt(cls, v, info: core_schema.FieldValidationInfo):
+    #     if info.data.get("init_image") is None and v:
+    #         msg = "You must set `init_image` if you want to use a mask"
+    #         raise ValueError(msg)
+    #     return v
 
-        return self
+    # @field_validator("model", mode="before")
+    # def set_default_diffusion_model(cls, v):
+    #     if v is None:
+    #         return config.DEFAULT_MODEL
 
-    @field_validator("height", "width")
-    def validate_image_size(cls, v, info: core_schema.FieldValidationInfo):
-        from imaginairy.model_manager import get_model_default_image_size
+    #     return v
 
-        if v is None:
-            v = get_model_default_image_size(info.data["model"])
+    # @field_validator("fix_faces_fidelity", mode="before")
+    # def validate_fix_faces_fidelity(cls, v):
+    #     if v is None:
+    #         return 0.2
 
-        return v
+    #     return v
 
-    @field_validator("caption_text", mode="before")
-    def validate_caption_text(cls, v):
-        if v is None:
-            v = ""
+    # @field_validator("sampler_type", mode="after")
+    # def validate_sampler_type(cls, v, info: core_schema.FieldValidationInfo):
+    #     from imaginairy.samplers import SamplerName
 
-        return v
+    #     if v is None:
+    #         v = config.DEFAULT_SAMPLER
 
-    @property
-    def prompts(self):
-        return self.prompt
+    #     v = v.lower()
+
+    #     if info.data.get("model") == "SD-2.0-v" and v == SamplerName.PLMS:
+    #         raise ValueError("PLMS sampler is not supported for SD-2.0-v model.")
+
+    #     if info.data.get("model") == "edit" and v in (
+    #         SamplerName.PLMS,
+    #         SamplerName.DDIM,
+    #     ):
+    #         msg = "PLMS and DDIM samplers are not supported for pix2pix edit model."
+    #         raise ValueError(msg)
+    #     return v
+
+    # @field_validator("steps")
+    # def validate_steps(cls, v, info: core_schema.FieldValidationInfo):
+    #     from imaginairy.samplers import SAMPLER_LOOKUP
+
+    #     if v is None:
+    #         SamplerCls = SAMPLER_LOOKUP[info.data["sampler_type"]]
+    #         v = SamplerCls.default_steps
+
+    #     return int(v)
+
+    # @model_validator(mode="after")
+    # def validate_init_image_strength(self):
+    #     if self.init_image_strength is None:
+    #         if self.control_inputs:
+    #             self.init_image_strength = 0.0
+    #         elif self.outpaint or self.mask_image or self.mask_prompt:
+    #             self.init_image_strength = 0.0
+    #         else:
+    #             self.init_image_strength = 0.2
+
+    #     return self
+
+
+@dataclass
+class ImaginePrompt:
+    prompt: list[WeightedPrompt]
+    negative_prompt: list[WeightedPrompt]
+    prompt_strength: float
+    init_image: InitImageConfig | None
+    width: int
+    height: int
+    seed: int
```

Comment:
Hmm this could be tricky.  I use `seed=None` to indicate that the user did not specify and thus we should randomly generate one.  Crucially though, if they pass in the same ImaginePrompt object, a second time, I think the reasonable expectation is that it generates a different image.

Thats why I have the "make_concrete_copy" that makes a copy of the prompt with finalized details like that.  

I'm sure there are other ways to handle this. open to ideas.